sentry-kotlin-multiplatform/src/commonMain/kotlin/io/sentry/kotlin/multiplatform/SentryBaseEvent.kt

```diff
@@ -0,0 +1,55 @@
+package io.sentry.kotlin.multiplatform
+
+import io.sentry.kotlin.multiplatform.protocol.Breadcrumb
+import io.sentry.kotlin.multiplatform.protocol.SentryId
+import io.sentry.kotlin.multiplatform.protocol.User
+
+/** Base class for all Sentry events. */
+public abstract class SentryBaseEvent(public open var eventId: SentryId = SentryId.EMPTY_ID) {
+
+    public open var release: String? = null
+    public open var environment: String? = null
+    public open var platform: String? = null
+    public open var user: User? = null
+    public open var serverName: String? = null
+    public open var dist: String? = null
+    protected var mutableContexts: Map<String, Any>? = null
+    public open val contexts: Map<String, Any>? get() = mutableContexts
+    private var mutableBreadcrumbs: MutableList<Breadcrumb>? = null
+    public open var breadcrumbs: List<Breadcrumb>?
+        get() = this.mutableBreadcrumbs
+        set(value) {
+            this.mutableBreadcrumbs = value?.toMutableList()
+        }
+
+    private var mutableTags: MutableMap<String, String>? = null
+    public open var tags: Map<String, String>?
+        get() = this.mutableTags
+        set(value) {
+            this.mutableTags = value?.toMutableMap()
+        }
+
+    public fun getTag(key: String): String? = this.mutableTags?.get(key)
+
+    public fun removeTag(key: String) {
+        this.mutableTags?.remove(key)
+    }
+
+    public fun setTag(key: String, value: String) {
+        if (mutableTags == null) {
+            this.mutableTags = HashMap()
+        }
+        this.mutableTags!![key] = value
+    }
+
+    public fun addBreadcrumb(breadcrumb: Breadcrumb) {
+        if (mutableBreadcrumbs == null) {
+            this.mutableBreadcrumbs = mutableListOf()
+        }
+        this.mutableBreadcrumbs?.add(breadcrumb)
+    }
```

Comment:
I rather prefer using `?` instead of `!!` otherwise you may crash the app.

Reply:
This code has concurrency issues if 2 threads are calling at the same time.
A simple way of solving this is always creating the list during declaration using a thread-safe type.

Reply:
Good point, didn't think of possible concurrency issues, will adjust it.

Reply:
Okay so one problem is that something like `ConcurrentHashMap` comes from the Java lib but I don't have access to that here since this is in the shared `commonMain` module.

Alternatives:
 - https://github.com/touchlab/Stately use something like this, concurrent collections specifically designed for KMP

Reply:
What you can do for now is, add a comment this is not thread-safe for every public list/map.
Raise an issue for making those things thread-safe, but let's not block this for now.